### PR TITLE
simplify check for math zone to make compatible with new vimtex syntax

### PR DESCRIPTION
**Pull request by Konfekt**
See also https://github.com/lervag/vimtex/issues/1880

### CODE REVIEW

File: tex.snippets [hunk 1]
```diff
@@ -1,22 +1,21 @@
...
 
 def math():
-	synstackids = vim.eval("synstack(line('.'), col('.') - (col('.')>=2 ? 1 : 0))")
+	synstackNames = vim.eval("map(synstack(line('.'), max([col('.') - 1, 1])), 'synIDattr(v:val, ''name'')')")
 	try:
-		first = next(i for i in reversed(synstackids) if i in texIgnoreMathZoneIds or i in texMathZoneIds)
-		return first != ignore
+		first = next(
+            i for i in reversed(synstackNames)
+            if (synstackNames[i].find(z) == 0 for z in texZones)
```
Comment (BertrandSim) [inline tex.snippets]: Instead of 
`if (synstackNames[i].find(z) == 0 for z in texZones)` ,
should it be 
`if any(i.find(z) == 0 for z in texZones)`  ?

Comment (Konfekt) [reply tex.snippets]: Yes, thank you. I am at loss why it just worked as intended, though the code contained leftovers such as using the list member `i` as its index.

**Comment by beinganukul** [comment]
there are bugs in this pull request

some snippets inside math() function refuse to work
``` 
163 context "math()"
164 snippet '([A-Za-z])(\d)' "auto subscript" wrA
165 `!p snip.rv = match.group(1)`_`!p snip.rv = match.group(2)`
166 endsnippet
```
this doesn't work 
but,
``` 163 #context "math()"
164 snippet '([A-Za-z])(\d)' "auto subscript" wrA
165 `!p snip.rv = match.group(1)`_`!p snip.rv = match.group(2)`
166 endsnippet
```
this works globally i don't really know how math function works but you guys can take a look and can reproduce this bug hopefully.

**Comment by Konfekt** [comment]
I cannot reproduce. I have been testing this PR in Vim `8.0.1568` on Linux using this abbreviation 

```
context "math()"
snippet '([A-Za-z])(\d)' "auto subscript" wrA
`!p snip.rv = match.group(1)`_`!p snip.rv = match.group(2)`
endsnippet
```

and, typing, say `a3`, in `$*$` (where `*` denotes the cursor position) expands to `a_3` whereas in `$text{*}$` it  stays `a3`

**Comment by beinganukul** [comment]
> I cannot reproduce. I have been testing this PR in Vim `8.0.1568` on Linux using this abbreviation
> 
> ```
> context "math()"
> snippet '([A-Za-z])(\d)' "auto subscript" wrA
> `!p snip.rv = match.group(1)`_`!p snip.rv = match.group(2)`
> endsnippet
> ```
> 
> and, typing, say `a3`, in `$*$` (where `*` denotes the cursor position) expands to `a_3` whereas in `$text{*}$` it stays `a3`

hey thank you for quick response.
i reinstalled vimplug with all other plugins and it worked after all without any issue.
hope gilles notices this pull request cause the script no longer functions properly in latest vimtex and thank you for your contribution have a nice day :)

**Comment by rednaz1337** [comment]
Can confirm that this PR works fine but the original code doesn't.

**Comment by isakhammer** [comment]
Can also confirm that this pull request works and the current master doesnt. 

Thanks for the fix!! Great job

**Comment by danielfalbo** [comment]
Seems like there's a much simpler solution but I haven't tried it yet:
```python3
def math():
    return vim.eval("vimtex#syntax#in_mathzone()") != "0"
```
.
Found [here](https://www.reddit.com/r/neovim/comments/kgszer/which_is_the_best_snippet_engine_and_why/ggi0sxu/?utm_source=reddit&utm_medium=web2x&context=3).

**Comment by Konfekt** [comment]
Yes, evaluating this`vimtex` function (from https://github.com/lervag/vimtex/pull/1884 ) achieves the same as this PR , but, well, depends on `vimtex`.

**Comment by gillescastel** [comment]
I updated the code a few days ago, making use of the vimtex functions:

https://github.com/gillescastel/latex-snippets/blob/baa1b42ed9558b4a7ba220c50c5046957719f898/tex.snippets#L1-L12

**Comment by BertrandSim** [comment]
Hello @gillescastel! Thanks for simplifying the code, it looks much cleaner for `vimtex` users. 

In `def env(name)`, I think we're missing a `y` somewhere in the return statement?

**Pull request closed without merging**